fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -709,10 +709,51 @@ struct CommitBatchContext {
 
 	std::set<Tag> getWrittenTagsPreResolution();
 
+	bool checkHotShards();
+
 private:
 	void evaluateBatchSize();
 };
 
+bool CommitBatchContext::checkHotShards() {
+	// removed expired hot shards
+	for (auto it = pProxyCommitData->hotShards.begin(); it != pProxyCommitData->hotShards.end();) {
+		if (now() > it->second) {
+			it = pProxyCommitData->hotShards.erase(it);
+		} else {
+			++it;
```

Comment:
is this list sorted by time? if so, we can break here.

Reply:
its not sorted by time.